services/JsonReader.class.php

```diff
@@ -0,0 +1,344 @@
+            // var_dump(implode("->" , $pathArray));
+            // echo "<br>";
+            // echo "<br>SubKey";
+            // var_dump($subKey);
+            // echo "<br>";
+            // echo "<br>Val";
+            // var_dump($val);
+            // echo "<br>";
+            // echo "<br>Arr";
+            // var_dump($arr);
+            // echo "<br>";
+            //Prevent people from adding to a Key Value pair into a non array type
+            // Example from Json Reference Object in Comments above
+            // $key = "data;_headers;newKey"
+            // This is not allowed because the Type of _header is a string,
+            // and you cannot have multiple key value objects in non array types such as int, string
+            
+
+            /*
+                if(isset($val)){
+                    // Check if whatever the value is: is it a multivalue object or just a single primitive type.                 
+                    if($v){echo"<br>Dumping count<br>";var_dump(count((array)$val ));
+                                var_dump(count((array)$val ) == 1);}
+                    if(count((array)$val ) == 1){
+                        echo "<br>";
+                        echo "<br>";
+                        echo "<br>";
+                        echo "<br>";
+                        throw new Exception("Cannot Add Multiple Objects To Non-Array Type ".get_class ($val).var_dump($val));
+                    }
+                }
+                **/
+            array_splice($pathArray, 0, 1);
+            $newKey = implode(";", $pathArray);
+            self::addToJsonMap($arr[$subKey], $newKey, $value);
+
+        }
+    }
+
+    /**
+     * Function: delete an Key -> Value from @param arr
+     * @param arr The Object we are deleting from.
+     * @param key prefix + keyName sperated by dilimiter ';'
+     *        @param prefix the path of the object you want to put `keyName` under
+     *            Ex using the Json Reference object above
+     *               if you want to add ['phoneNumber'] => ['2020020202'] under `indexes`
+     *               prefix = `data;indexes;' and key = `phoneNumber`
+     *              * Notice Prefix end has ;
+     *            Ex adding to the main object
+     *              prefix =  "" key = "phoneNumber"
+     *
+     */
+    function deleteFromJsonMap(&$arr, $key)
+    {
+        $pathArray = explode(";", $key);
+        $size = sizeof($pathArray);
+        print ("Size of Key to delete is: '" . $key . "' is " . $size . "\n");
+        var_dump($pathArray);
+        if (sizeof($pathArray) == 1)
+        {
+            // No need to traverse
+            settype($arr, "array");
+
+            if (!isset($arr[$key]))
+            {
+                echo "4";
+                return new Exception("Trying To Delete Something That Does Not Exist");
+                echo "3";
+            }
+            echo "2";
+            //Not sure if this is working

```

Comment:
//Not sure if this is working

are you sure it is working now?
might want to do unit testing on this? it might be a good idea and check in VScode for the code coverage. so you will know what methods work completely, and which do not work